src/stan/variational/families/normal_fullrank.hpp

```diff
@@ -49,12 +49,15 @@ namespace stan {
 
     public:
       // Constructors
+
+      // Constructs a zero-filled variational distribution
       explicit normal_fullrank(size_t dimension) :
         dimension_(dimension) {
         mu_     = Eigen::VectorXd::Zero(dimension_);
-        L_chol_  = Eigen::MatrixXd::Identity(dimension_, dimension_);
+        L_chol_ = Eigen::MatrixXd::Zero(dimension_, dimension_);
```

Comment:
These shold all be initializers, as in `mu_(Eigen::VectorXd::Zero(dimension))`.

You should prefer to use local variables like `dimension` instead of member variables like `dimension_`.  Just make sure to put them in the same order as the declaration, where dimension is last.

I'm like a broken record in this regard, but wherever possible, it's better to have immutable objects (const member variables in C++ terms).  The main reason is that it's easier to make sure the class is in a sound state when it's constructed.  If you have setters, you need to be extra careful where you validate that a class is finally ready to go.  Here, the class isn't ready to go presumably until L_chol_ is set to something more useful.

Why set L_chol_ to the identity?

We would very strongly prefer doxygen doc on the classes and functions, not line comments.  No need to add comments like "Constructors" --- follow the Google style guide and assume the people reading your code know C++ and can tell that's a constructor by its name.